cmd/cli/node/list.go

```diff
@@ -42,22 +44,24 @@ func NewListCmd() *cobra.Command {
 		Use:   "list",
 		Short: "List info of network nodes. ",
 		Args:  cobra.NoArgs,
-		Run:   o.run,
+		RunE:  o.run,
 	}
 	nodeCmd.Flags().StringSliceVar(&o.ColumnGroups, "show", o.ColumnGroups,
 		fmt.Sprintf("What column groups to show. Zero or more of: %q", maps.Keys(toggleColumns)))
 	nodeCmd.Flags().StringVar(&o.Labels, "labels", o.Labels,
 		"Filter nodes by labels. See https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/ for more information.")
 	nodeCmd.Flags().AddFlagSet(cliflags.ListFlags(&o.ListOptions))
 	nodeCmd.Flags().AddFlagSet(cliflags.OutputFormatFlags(&o.OutputOptions))
+	nodeCmd.Flags().StringVar(&o.FilterByApproval, "filter-approval", o.FilterByApproval,
+		fmt.Sprintf("Filter nodes by approval. One of: %q", filterApprovalValues))
 	nodeCmd.Flags().StringVar(&o.FilterByStatus, "filter-status", o.FilterByStatus,
 		fmt.Sprintf("Filter nodes by status. One of: %q", filterStatusValues))
 
 	return nodeCmd
 }
 
 // Run executes node command
-func (o *ListOptions) run(cmd *cobra.Command, _ []string) {
+func (o *ListOptions) run(cmd *cobra.Command, _ []string) error {
```

Comment:
if we are going to return an error from this method can we replace all the `util.Fatal` calls while we are at it? (be wary  this might break some tests that (annoyingly) override the `util.Fatal` field :disappointed:)

Reply:
Can definitely add a ticket to address this, yeah. There's a mix of approaches atm.

Reply:
Added https://github.com/bacalhau-project/bacalhau/issues/3764